doozerlib/cli/release_gen_payload.py

```diff
@@ -502,155 +506,131 @@ def _inconsistency_annotation(self, inconsistencies):
+                # name explicitly, it will take precedence over any other entries
+                # https://issues.redhat.com/browse/ART-2823
+                # This was tag not explicitly declared, so ignore the duplicate image.
+                continue
 
-def default_is_base_name(runtime: Runtime):
-    version = runtime.get_minor_version()
-    if runtime.assembly is None or runtime.assembly == 'stream':
-        return f'{version}-art-latest'
-    else:
-        return f'{version}-art-assembly-{runtime.assembly}'
+            archive_inspector = build_inspector.get_image_archive_inspector(brew_arch)
 
+            if not archive_inspector:
+                # This is no build for this CPU architecture for this build. Don't worry yet,
+                # it may be carried by an -alt image or not at all for non-x86 arches.
+                members[tag_name] = None
+                continue
 
-def default_is_base_namespace():
-    return "ocp"
+            members[tag_name] = archive_inspector
 
+        return members
 
-def is_name_and_space(base_name, base_namespace, brew_arch, private):
-    arch_suffix = go_suffix_for_arch(brew_arch)
-    priv_suffix = "-priv" if private else ""
-    name = f"{base_name}{arch_suffix}{priv_suffix}"
-    namespace = f"{base_namespace}{arch_suffix}{priv_suffix}"
-    return name, namespace
+    @staticmethod
+    def _check_nightly_consistency(assembly_inspector: AssemblyInspector, nightly: str, arch: str) -> List[str]:
+        runtime = assembly_inspector.runtime
+
+        def terminal_issue(msg: str):
+            return [msg]
+
+        issues: List[str]
+        runtime.logger.info(f'Processing nightly: {nightly}')
+        major_minor, brew_cpu_arch, priv = isolate_nightly_name_components(nightly)
+
+        if major_minor != runtime.get_minor_version():
+            return terminal_issue(f'Specified nightly {nightly} does not match group major.minor')
+
+        rc_suffix = go_suffix_for_arch(brew_cpu_arch, priv)
+
+        retries: int = 3
+        release_json_str = ''
+        rc = -1
+        pullspec = f'registry.ci.openshift.org/ocp{rc_suffix}/release{rc_suffix}:{nightly}'
+        while retries > 0:
+            rc, release_json_str, err = exectools.cmd_gather(f'oc adm release info {pullspec} -o=json')
+            if rc == 0:
+                break
+            runtime.logger.warn(f'Error accessing nightly release info for {pullspec}:  {err}')
+            retries -= 1
+
+        if rc != 0:
+            return terminal_issue(f'Unable to gather nightly release info details: {pullspec}; garbage collected?')
+
+        release_info = Model(dict_to_model=json.loads(release_json_str))
+        if not release_info.references.spec.tags:
+            return terminal_issue(f'Could not find tags in nightly {nightly}')
+
+        issues: List[str] = list()
+        payload_entries: Dict[str, PayloadGenerator.PayloadEntry] = PayloadGenerator.find_payload_entries(assembly_inspector, arch, '')
```

Comment:
Would like to cache the result of `find_payload_entries` rather than call it twice.

Reply:
I think the only heavy lift in this code path is populating `assembly_inspector.get_group_release_images`, which is cached. If we decide we should cache the entries anyway, I would want to move `find_payload_entries` into the assembly_inspector to cache cleanly and that starts to muddy the AssemblyInspector / PayloadGenerator abstractions.